### PR TITLE
Support the use of custom authentication mechanism, timeouts, and other `httpx` parameters in Python Client

### PR DESCRIPTION
**Pull request by valgai**
## Description

Allow users of the `gradio_client` library to forward authentication mechanism to `httpx` calls using a new `httpx_auth` 
parameter in `gradio_client.client.Client` signature.

Closes: #8754 
Closes: https://github.com/gradio-app/gradio/issues/6049

## 🎯 PRs Should Target Issues

Before your create a PR, please check to see if there is [an existing issue](https://github.com/gradio-app/gradio/issues) for this change. If not, please create an issue before you create this PR, unless the fix is very small. 

Not adhering to this guideline will result in the PR being closed. 

## Tests

1. PRs will only be merged if tests pass on CI. To run the tests locally, please set up [your Gradio environment locally](https://github.com/gradio-app/gradio/blob/main/CONTRIBUTING.md) and run the tests: `bash scripts/run_all_tests.sh`

2. You may need to run the linters: `bash scripts/format_backend.sh` and `bash scripts/format_frontend.sh`

### CODE REVIEW

File: client/python/gradio_client/client.py [hunk 1]
```diff
@@ -79,6 +80,7 @@ def __init__(
         max_workers: int = 40,
         verbose: bool = True,
         auth: tuple[str, str] | None = None,
+        httpx_auth: AuthTypes | None = None,
```
Comment (freddyaboulton) [inline client/python/gradio_client/client.py:83]: Should the type be `httpx.Auth`?

**Review by freddyaboulton — APPROVED** [review]
Looks good @valgai - just left a minor suggestion to not import from private `_auth` module

**Comment by valgai** [comment]
Yes, I agree it's ugly. The `httpx.Auth` type is one member of the union represented by `httpx._types.AuthTypes`.
So we do have:

```
class Auth: ...
```

and

```python
AuthTypes = Union[
    Tuple[Union[str, bytes], Union[str, bytes]],
    Callable[["Request"], "Request"],
    "Auth",
]
``` 

However, through `httpx.BasicAuth` and `httpx.auth_.FunctionAuth` types, instances of `Auth` can be created using a `Tuple[Union[str, bytes], Union[str, bytes]]` or a `Callable[["Request"], "Request"]` as an argument. Only `FunctionAuth` is not exported publicly.

**Comment by abidlabs** [comment]
The `httpx.Client` class has [a lot of kwargs](https://www.python-httpx.org/api/#client) and instead of adding them individually to our `Client` class, what if we just add a `client_kwargs: dict[str, Any]` argument and forward them to the `httpx.Client` constructor?

**Comment by abidlabs** [comment]
@valgai would you be able to update your PR with the suggestion above? This way will better from a maintenance perpsective

**Comment by valgai** [comment]
Sure! As `client_kwargs` is being forwarded to `httpx.stream`, `httpx.post`, `httpx.get` in addition to `httpx.Client`, maybe we could name it `httpx_kwargs` instead?

**Comment by abidlabs** [comment]
sounds good to me, let's change to `httpx_kwargs`

**Review by abidlabs — APPROVED** [review]
lgtm thanks @valgai. going to add a test and we should be good to merge

**Comment by valgai** [comment]
Thanks!

**Pull request merged**